src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java

```diff
@@ -980,6 +993,42 @@ protected void setAttributes(Section mainSection) throws MojoExecutionException,
     }
 
     /**
+     * Determine the minimum Java version this plugin requires from HPI plugin configuration or other POM entries.
+     *
+     * @return A Java version such as <code>1.7</code> or <code>1.8</code>, or <code>null</code> if it could not be determined.
+     */
+    private String determineMinimumJavaVersion() {
+        // prefer explicitly defined minimum Java version
+        String javaVersion = this.javaVersion;
+
+        // next stop, 2.x plugins parent POM property
+        if (javaVersion == null) {
+            // plugin POM convention
+            javaVersion = this.project.getProperties().getProperty("java.level");
```

Comment:
No, do not do it this way.

Define a mojo property with no default. Then patch `plugin-pom` to configure it with `${java.level}`.

Reply:
done